### PR TITLE
Migrate to python3, and fix warnings/errors picked up by GCC 10.2

### PR DESCRIPTION
**Pull request by gemarcano**
Let me know if you'd like for me to split the request into two.

Most python3 changes were done by 2to3. I did some manual fixes for some issues I saw while doing a manual review of the changes.

I'm not 100% sure I put `return`/`break` in the right places to fix the compiler complaints (mostly complaining about falling through).

### CODE REVIEW

File: simulator/core/gdb.c [hunk 1]
```diff
@@ -400,6 +400,7 @@ static bool _wait_for_gdb(void) {
 			// kill
 			INFO("Killed by remote debugger, dying\n");
 			sim_terminate(true);
+			break;
```
Comment (ppannuto) [inline simulator/core/gdb.c:403]: Ahh... nice catch. One of those 'not technically possible' things – once upon a time, `sim_terminate` was marked `__attribute__ ((noreturn))` so this wasn't actually a fallthrough (and with the argument `true`, it still isn't, but the compiler can't know that). Later on I added an option to trap on termination to allow a debugger to hook and analyze the final core

File: simulator/core/simulator.c [hunk 2]
```diff
@@ -294,9 +295,11 @@ static void _shell(void) {
 				return _shell();
 			}
 		}
+		break;
```
Comment (ppannuto) [inline simulator/core/simulator.c:298]: I don't think this `break` should be here, given the comment on line 279. Not the most intuitive control flow, but I think what that does is try to get a valid filename in the first half of the `r` case. If there is a file, then the second half of the `r` case will `return` (so fall through doesn't matter). If there isn't a file, then it relies on falling through the cases to get to the `h(elp)` case.

**Review by ppannuto — COMMENTED** [review]
Awesome! Thanks for all of these updates. I pushed a few small additional changes and have one question about whether an added break is correct. Other than that I think this is good to go.

File: simulator/core/gdb.c (hunk 1 above)
Comment (gemarcano) [reply simulator/core/gdb.c:403]: Odd. Usually GCC can catch things like this especially if `should_exit` is defined as a constant (which it is in this case). Maybe the C library `exit` isn't defined as noreturn?

File: simulator/core/simulator.c (hunk 2 above)
Comment (gemarcano) [reply simulator/core/simulator.c:298]: OK, that makes sense, and yes, I see the fallthrough in the code (I almost wonder if a `goto` might help in making this cleaner? That's a problem for another time).

**Review by ppannuto — APPROVED** [review]

**Pull request merged**